agent/api/eni/eni.go

```diff
@@ -83,28 +83,49 @@ func (eni *ENI) GetPrimaryIPv4Address() string {
 		}
 	}
 
-	return primaryAddr
+	return eni.getIPAddressWithBlockSize(primaryAddr, true)
 }
 
-// GetIPV6Addresses returns a list of ipv6 addresses allocated to the ENI
+// GetIPV6Addresses returns the ipv6 addresses allocated to the ENI.
 func (eni *ENI) GetIPV6Addresses() []string {
 	var addresses []string
 	for _, addr := range eni.IPV6Addresses {
-		addresses = append(addresses, addr.Address)
+		addresses = append(addresses, eni.getIPAddressWithBlockSize(addr.Address, false))
 	}
 
 	return addresses
 }
 
-// GetHostname returns the hostname assigned to the ENI
+// GetIPAddresses returns all the ip addresses allocated to the ENI.
+func (eni *ENI) GetIPAddresses() []string {
+	ipv4Addr := eni.GetPrimaryIPv4Address()
```

Comment:
Why are we returning only the primary IPv4 address? The method name suggests it is supposed to return all IP addresses. I am guessing the reason is that GetIPv4Addresses doesn't include the prefix length, but that's an implementation detail that shouldn't leak to method behavior.

Reply:
removed this and updated to use the method you added